----------

<center>
<h1><b>T-Test</b></h1>
</center>

---------------

A T-test is a statistical test used to determine if there is a significant difference between the means of two groups. Here’s a simple breakdown:

### Types of T-Tests
1. **One-Sample T-Test**:
   - **Purpose**: To determine if the mean of a single sample is significantly different from a known or hypothesized population mean.
   - **Example**: Testing if the average height of students in a class is different from the national average height.

2. **Independent (Two-Sample) T-Test**:
   - **Purpose**: To compare the means of two independent groups to see if they are significantly different from each other.
   - **Example**: Comparing the performance of two website versions after A/B testing to determine which one performs better.
     
3. **Paired (Dependent) T-Test**:
   - **Purpose**: To compare means from the same group at different times (e.g., before and after a treatment) or under two different conditions.
   - **Example**: Testing the effectiveness of a new drug by comparing the health metrics of the same patients before and after treatment.

### Suitable Situations for Usage
- **One-Sample T-Test**:
  - When you have a single sample and want to compare it against a known value.
  - Example: Testing if the average daily intake of sugar in a sample group is different from the recommended daily intake.

- **Independent T-Test**:
  - When comparing two different groups to see if they differ significantly in a particular metric.
  - Example: Comparing the performance of two different marketing strategies.

- **Paired T-Test**:
  - When you have repeated measurements or matched pairs.
  - Example: Assessing the impact of a training program by comparing pre-training and post-training test scores of the same employees.

### Situations Where T-Test Cannot Be Used
- **Non-Normal Distribution**: T-tests assume the data follows a normal distribution. If the data is not normally distributed, other tests like the Mann-Whitney U test (for independent samples) or the Wilcoxon signed-rank test (for paired samples) might be more appropriate.
- **More Than Two Groups**: T-tests are designed for comparisons between two groups. For comparisons involving more than two groups, ANOVA (Analysis of Variance) should be used.
- **Small Sample Sizes**: While t-tests can be used with small sample sizes, the results may not be reliable. Larger sample sizes generally give more robust results.
- **Outliers**: T-tests are sensitive to outliers, which can significantly affect the results. If outliers are present, they should be addressed before conducting a t-test.

### Limitations of T-Tests
- **Assumption of Normality**: T-tests assume that the data are normally distributed. This may not always be the case, especially with small sample sizes.
- **Homogeneity of Variances**: For independent t-tests, there is an assumption that the variances of the two groups are equal. If this assumption is violated, the results might be invalid.
- **Sensitivity to Outliers**: Outliers can greatly affect the results of a t-test. Care should be taken to identify and manage outliers.
- **Binary Comparison**: T-tests are limited to comparing the means of two groups only. For more than two groups, other statistical methods like ANOVA are needed.
- **Sample Size**: Small sample sizes can lead to unreliable results. Larger samples provide more reliable and robust outcomes.

For further reading on T-tests, you can refer to [Khan Academy](https://www.khanacademy.org/math/statistics-probability/inference-about-populations/one-sample-t-test/v/t-statistic-confidence-interval) or [Statistics Solutions](https://www.statisticssolutions.com/manova-analysis-anova/paired-sample-t-test/).

## T- test Calculation

In [1]:
# import needed libraries
import numpy as np
from scipy import stats

### One-Sample T-Test
Example: Imagine we want to see if kids in a class eat more sugar than the amount their doctor says is okay. The doctor recommends that kids should eat 50 grams of sugar a day.

**Generate Data and Calculate T-Test**

In [2]:
# Generate example data
np.random.seed(0)  # setting the random number seed
sugar_intake = np.random.normal(55, 10, 30)  # mean=55, std=10, n=30

# Perform one-sample t-test
t_stat, p_value = stats.ttest_1samp(sugar_intake, 50)  # 50 (gram) is the recommended daily sugar intake 

t_stat, p_value


(4.693360062287237, 5.944474834473945e-05)

### One-Sample T-Test Interpretation
- **T-Statistic**: 4.693
- **P-Value**: 0.0000594 (approximately)

We measured how much sugar 30 kids ate and found out that, on average, they ate 55 grams. We used a special tool (the t-test) to see if this average amount (55 grams) was really different from the doctor's recommendation (50 grams) or if it was just by chance.

#### What We Found:
1. **The T-Statistic (4.693)**: This number tells us how big the difference is between what the kids actually ate (55 grams) and what the doctor recommended (50 grams). A bigger number means a bigger difference.

2. **The P-Value (0.0000594)**: This very small number tells us the likelihood that this difference happened just by chance. Since this number is much smaller than 0.05 (which is like saying 5 out of 100), it means the difference is very unlikely to be just by chance.

**In Simple Words**:
It's almost certain that kids in this class really do eat more sugar than the doctor says is okay, not just by luck or random chance. It's like if you find more candies in one bag than another bag, and you're very sure it's not just by accident.

So, we can say with a lot of confidence that the kids are eating more sugar than they should be. That will lead us to the independent t-test

____

## Independent T-Test
Example: Imagine you have two different versions of a webpage (Version A and Version B), and you want to see if visitors spend more time on one version than the other. You measured how long 50 visitors spent on each version.

**Generate Data and Calculate T-Test**

In [3]:
# Generate example data
np.random.seed(42)  # setting the random number seed
time_spent_A = np.random.normal(5, 1, 50)  # mean=5 minutes, std=1 minute, n=50
time_spent_B = np.random.normal(5.5, 1, 50)  # mean=5.5 minutes, std=1 minute, n=50

# Perform independent t-test
t_stat, p_value = stats.ttest_ind(time_spent_A, time_spent_B)

t_stat, p_value


(-4.108723928204804, 8.261945608702763e-05)

### Independent T-Test Interpretation

- **T-Statistic**: -4.109 (approximately)
- **P-Value**: 0.0000826 (approximately)

**What We Found**:
1. **The T-Statistic (-4.109)**: This number tells us there is a big difference in the time visitors spent on the two versions. Since the number is negative, it means visitors spent less time on Version A compared to Version B.
2. **The P-Value (0.0000826)**: This tiny number tells us that the difference in time spent is very, very unlikely to be just by chance. Since this number is much smaller than 0.05, it means the difference is statistically significant.

**In Simple Words**:
We found that visitors spend much less time on Version A of the webpage compared to Version B. This result is almost certainly true and not just a lucky coincidence. It's like finding out that one webpage is much less engaging because visitors stayed on it for a much shorter time, and we are very sure it's not just by accident.

So, we can be very confident that Version B of the webpage keeps users engaged for a longer time compared to Version A.

Let's now look into the third type of t-test, which is paired t-test.

----

## Paired T-Test
Example: Assessing the impact of a training program by comparing pre-training and post-training test scores of the same employees.

**Generate Data and Calculate T-Test**

In [4]:
# Generate example data
np.random.seed(2)  # setting the random number seed
pre_training = np.random.normal(65, 8, 30)  # mean=65, std=8, n=30
post_training = np.random.normal(70, 7, 30)  # mean=70, std=7, n=30

# Perform paired t-test
t_stat, p_value = stats.ttest_rel(pre_training, post_training)

t_stat, p_value


(-4.732082893741267, 5.339064806342024e-05)

### Paired T-Test Interpretation
- **T-Statistic**: -4.732 (approximately)
- **P-Value**: 0.0000534 (approximately)

**What We Found**:
1. **The T-Statistic (-4.732)**: This number tells us that there is a big difference between the scores before training and after training the employees. Since the number is negative, it means the scores were lower before the training and got higher after training the employees.
2. **The P-Value (0.0000534)**: This very small value tells us that the improvement in scores is almost certainly not just by chance. Since this number is much smaller than 0.05, it means the improvement is statistically significant.

**In Simple Words**:
We found that employees really did get much better after the training. This result is almost certainly true and not just a lucky coincidence. 

So, we can be very confident that the training program helped the employees improve their test scores significantly.

<center>
<h2><b>Thank You</b></2>
</center>


**Contact**: [owabumoye@gmail.com](mailto:owabumoye@gmail.com)